src/backend/distributed/citus--8.3-1--8.4-1.sql

```diff
@@ -1,6 +1,78 @@
 /* citus--8.3-1--8.4-1 */
 
 /* bump version to 8.4-1 */
+CREATE SCHEMA IF NOT EXISTS citus_internal;
+
+CREATE OR REPLACE FUNCTION citus_internal.prepare_pg_upgrade_pg_dist_object()
+  RETURNS void
+  LANGUAGE C STRICT
+  AS 'MODULE_PATHNAME', 'citus_prepare_pg_upgrade_pg_dist_object';
+
+CREATE OR REPLACE FUNCTION citus_internal.finish_pg_upgrade_pg_dist_object()
+  RETURNS void
+  LANGUAGE C STRICT
+  AS 'MODULE_PATHNAME', 'citus_finish_pg_upgrade_pg_dist_object';
+
+CREATE TABLE citus.pg_dist_object (
+    classid oid NOT NULL,
+    objid oid NOT NULL,
+    identifier text DEFAULT NULL -- used to store a stable identifier during pg_upgrade
+);
+
+CREATE INDEX pg_dist_object_classid_objid_index ON
+    citus.pg_dist_object USING btree(classid, objid);
+
+GRANT USAGE ON SCHEMA citus TO public;
```

Comment:
Would be good to move the 6 functions in the citus schema to the citus_internal schema.

Reply:
This is done in https://github.com/citusdata/citus/pull/2882/commits/ab713c126bf089b8e6a377232a5cc7fc4451169f

Would this make sense to keep in this PR or should we move it to a separate PR for auditability?

Reply:
I think this PR is fine as long as it's a separate commit.